packages/client/hmi-client/src/services/operator-bitmask.ts

```diff
@@ -0,0 +1,35 @@
+// These values represent different interaction states on an operator node
+// The operator header, border, etc. will be styled differently depending on which states are active
+enum Status {
+	Found = 0x0001,
+	Hover = 0x0010,
+	Focus = 0x0100,
+	Drag = 0x1000
+}
+
+const addState = (status: number, state: Status) => {
+	status |= state;
+	return status;
+};
+const removeState = (status: number, state: Status) => {
+	status &= ~state;
+	return status;
+};
+const isStateActive = (status: number, state: Status) => status & state;
```

Comment:
Not to be pedantic, but I think you reverse the meaning of `state` and `status`?
```suggestion
const addState = (state: number, status: Status) => {
	state |= status;
	return state;
};
const removeState = (state: number, status: Status) => {
	state &= ~status;
	return state;
};
const isStateActive = (state: number, status: Status) => state & status;
```